jina/flow/base.py

```diff
@@ -708,6 +708,10 @@ def add(
             kwargs, parser, True, fallback_parsers=FALLBACK_PARSERS
         )
 
+        # grpc data runtime does not support sharding at the moment
+        if args.grpc_data_requests and kwargs.get('shards') is not None:
```

Comment:
If None is fine, then shouldnt check should be:
`if args.grpc_data_requests and kwargs.get('shards') is not None and kwargs.get('shards') > 1:`

Reply:
yes indeed, to make the if a bit shorter you can provide a default value to the .get() function in case its not existing:
`kwargs.get('shards', 1) `

Reply:
This could work. But what about cases `Flow(grpc_data_requests=True).add(shards=None) ` because it overrides defailt value 1 and gives error during if-condition.

Reply:
ok, true so I guess we always need to check for None then as well.